frontend/app/components/Home.tsx

```diff
@@ -25,34 +28,44 @@ const filterTools = (tools: Tool[], filter: string): Tool[] => {
 };
 
 function Home () {
-  const [tools, setTools] = useState<null | Tool[]>(null);
+  const [tools, setTools] = useState<undefined | Tool[]>(undefined);
+  const [showRefreshAlert, setShowRefreshAlert] = useState<undefined | boolean>(undefined);
+
+  const { isLoading: getToolsLoading, error: getToolsError } = useQuery('getTools', api.getTools, {
+    onSuccess: (data) => setTools(data)
+  });
+
+  const onToolUpdate = (newTool: Tool) => {
+    /*
+    Creates new array with newTool replacing its previous version;
+    Uses function inside setState hook to handle overlapping requests
+    */
+    setTools((oldTools) => {
+      if (oldTools === undefined) throw Error('Expected tools variable to be defined!');
```

Comment:
So, this isn't an API error. `createErrorMessage` is intended to process a `Response` object and return a string summarizing what happened with an API error.

I don't ever expect this to happen; this function should only be called inside of a `ToolCard`, which would need to have tool data for it to work. However, because `tools` in the `useState` hook starts off as `undefined`, it is possible if the function is used incorrectly. To satisfy TypeScript, I am throwing an error here. If you think of a better solution, please let me know.